### PR TITLE
Automated release draft 1

### PR DESCRIPTION
**Pull request by xorima**
## Description

Blog post for Automated Release process

### Issues Resolved

[List any existing issues this PR resolves]

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Review by bmhughes — DISMISSED** [review]

File: content/docs/auto-releasing-sous-cookbooks.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+While these cookbooks are great, when we typically take over management of the cookbook repos they are not in the best of states, and there is significant work to bring them up to our standards.
+
+We also have all our existing cookbooks (of which there are over 70) and as our standards change, we need to deploy these changes across all of our repositories.
+
+## Enter Automation
+
+Earlier in the year we build the [Xorimabot backend suite](https://sous-chefs.org/docs/managing-cookbooks-at-scale/)
+
+We now have cookbooks which are constantly getting updated but moved onto the next problem, releasing these is a manual process which takes time from our variour contributors (Which normally is [Ramereth](https://github.com/Ramereth)).
```
Comment (ramereth) [inline content/docs/auto-releasing-sous-cookbooks.md]:
```suggestion
We now have cookbooks which are constantly getting updated but moved onto the next problem, releasing these is a manual process which takes time from our various contributors (Which normally is [Ramereth](https://github.com/Ramereth)).
```

File: content/docs/auto-releasing-sous-cookbooks.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Here we truely believe all members of the community should be able to release cookbooks as long as the normal checks (CI, Review) are passed so we did what any DevOps team would do, we built automation.
+
+### Architecture
+
+We started to partition the problem into smaller parts, below you can see the architecute we came up with in the end.
```
Comment (ramereth) [inline content/docs/auto-releasing-sous-cookbooks.md]:
```suggestion
We started to partition the problem into smaller parts, below you can see the architecture we came up with in the end.
```

File: content/docs/auto-releasing-sous-cookbooks.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+We also have all our existing cookbooks (of which there are over 70) and as our standards change, we need to deploy these changes across all of our repositories.
+
+## Enter Automation
+
+Earlier in the year we build the [Xorimabot backend suite](https://sous-chefs.org/docs/managing-cookbooks-at-scale/)
+
+We now have cookbooks which are constantly getting updated but moved onto the next problem, releasing these is a manual process which takes time from our variour contributors (Which normally is [Ramereth](https://github.com/Ramereth)).
+
+Here we truely believe all members of the community should be able to release cookbooks as long as the normal checks (CI, Review) are passed so we did what any DevOps team would do, we built automation.
```
Comment (ramereth) [inline content/docs/auto-releasing-sous-cookbooks.md]:
```suggestion
Here we truly believe all members of the community should be able to release cookbooks as long as the normal checks (CI, Review) are passed so we did what any DevOps team would do, we built automation.
```

File: content/docs/auto-releasing-sous-cookbooks.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+### Architecture
+
+We started to partition the problem into smaller parts, below you can see the architecute we came up with in the end.
+These applications currently run on Kubernetes, the aim is to migrate them to AWS Lambda simply due to the costs of running this solution. If you would like to aid us with covering the costs of sous-chefs donations would be welcome in our [Open Collective](https://opencollective.com/sous-chefs)
+
+![Sous-Chefs auto release architecture](../images/automation-architecture.png)
+
+In this architecture we have applications that cover the various problems we forsaw, these are covered in more detail below.
```
Comment (ramereth) [inline content/docs/auto-releasing-sous-cookbooks.md]:
```suggestion
In this architecture we have applications that cover the various problems we foresaw, these are covered in more detail below.
```

File: content/docs/auto-releasing-sous-cookbooks.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+[Cookbook Supermarket Uploader](https://github.com/Xorima/cookbook_supermarket_uploader) runs when a deployment is requested. It does the following:
+
+- Checks the Deployment was created by the same user it is running as
+- Clones the git code based on the tag which is pushed when a release is created
+- Runs a `knife supermarket share` to upload this to the chef supermarket
+- Posts back a sucessful or failed deployment to Github's API
+
+## Slack Notifier
+
+[Deployment Status Slack Notifier](https://github.com/Xorima/deployment_status_slack_notifier) runs when a [Deployment Status](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-deployment-statuses) is updated to Success of Failure. It then route the message to the [Chef Community Slack](https://community-slack.chef.io/). It gets the release notes from the payload of the deployment and posts them to `#Sous-Chefs` and `#Anouncements`
```
Comment (ramereth) [inline content/docs/auto-releasing-sous-cookbooks.md]:
```suggestion
[Deployment Status Slack Notifier](https://github.com/Xorima/deployment_status_slack_notifier) runs when a [Deployment Status](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-deployment-statuses) is updated to Success of Failure. It then route the message to the [Chef Community Slack](https://community-slack.chef.io/). It gets the release notes from the payload of the deployment and posts them to `#Sous-Chefs` and `#Announcements`
```

**Review by ramereth — DISMISSED** [review]
Typos. Should I be concerned about the build failure on circleci?

File: content/docs/auto-releasing-sous-cookbooks.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+We now have cookbooks which are constantly getting updated but moved onto the next problem, releasing these is a manual process which takes time from our variour contributors (Which normally is [Ramereth](https://github.com/Ramereth)).
+
+Here we truely believe all members of the community should be able to release cookbooks as long as the normal checks (CI, Review) are passed so we did what any DevOps team would do, we built automation.
+
+### Architecture
+
+We started to partition the problem into smaller parts, below you can see the architecute we came up with in the end.
+These applications currently run on Kubernetes, the aim is to migrate them to AWS Lambda simply due to the costs of running this solution. If you would like to aid us with covering the costs of sous-chefs donations would be welcome in our [Open Collective](https://opencollective.com/sous-chefs)
+
+![Sous-Chefs auto release architecture](../images/automation-architecture.png)
```
Comment (ramereth) [inline content/docs/auto-releasing-sous-cookbooks.md]:
```suggestion
![Sous-Chefs auto release architecture](/images/automation-architecture.png)
```

**Comment by xorima** [comment]
> Typos. Should I be concerned about the build failure on circleci?

I have no idea, this is the one ci setup I dont know

**Comment by ramereth** [comment]
> > Typos. Should I be concerned about the build failure on circleci?
> 
> I have no idea, this is the one ci setup I dont know

Yeah it looks like the newer version of hugo doesn't like the theme we use. It needs to be fixed before this can be merged.

**Pull request merged**